### ISSUE TITLE
Internal evaluation error: context is missing for node 1

### ISSUE THREAD

**Issue by opax** [eXist-db/exist#689]
I found a really strange bug that is reproducible with my exist-2.2 installation:

The query

```
declare function local:f() {
    xmldb:xcollection("/db/tmp")//html/body[table/@border = '1']
};
map { 'bla' := local:f() }
```

exits with the error message "context is missing for node ...", assuming a matching document exists in `/db/tmp`. 

It seems like every element in the query is necessary to reproduce the bug - the function must be called from within a map constructor, simply calling the function works. The simpler query `xmldb:xcollection(...)//body[table/@border='1']` inside the function works. Even the query `xmldb:xcollection(...)//body[../self::html][table/@border='1']` does not trigger the bug.

**Comment by opax** [comment]
FWIW, the same query worked without problems in eXist-db 2.1.

**Comment by opax** [comment]
Self-contained version of bug, works in eXide:

    xquery version "3.0";
    declare variable $collection-uri := "/db/issue689";

    declare function local:setup() {
        let $c := tokenize($collection-uri, '/')
        return
        xmldb:create-collection(string-join($c[position() lt last()], '/'), $c[last()]),
        xmldb:store($collection-uri, "bla.xml", <a><b><c d="1"/></b></a>)
    };

    declare function local:f() {
        xmldb:xcollection($collection-uri)//a/b[c/@d = "1"]
    };

    local:setup(),
    map { 'bla' := local:f() }

**Comment by dizzzz** [comment]
why do you use `xcollection()` in stead of `collection()` ?

**Comment by opax** [comment]
I tend to prefer `xcollection()` over `collection()` whenever I know that the relevant documents can't be in subcollections, so I do not have to bother about whether subcollections are actually present or not. Defensive programming against changes in the database layout :)

It is not relevant for the bug, though: The error is triggered with `collection()` as well as with `xcollection()`.

**Comment by opax** [comment]
The bug seems to be fixed in eXist-db 3.2.0 - I could not reproduce it any more.

**Comment by dizzzz** [comment]
Thnx for verifying
